tests/data/modules_v2/refs/inmanta_plugins/refs/dc.py

```diff
@@ -0,0 +1,78 @@
+import dataclasses
+from inmanta.agent.handler import LoggerABC
+from inmanta.plugins import plugin
+from inmanta.references import reference, Reference
+
+
+@dataclasses.dataclass(frozen=True, kw_only=True)
+class DataclassABC:
+    ...
+
+
+@dataclasses.dataclass(frozen=True, kw_only=True)
+class NoRefsDataclass(DataclassABC):
+    non_ref_value: str = "Hello World!"
+
+
+@dataclasses.dataclass(frozen=True, kw_only=True)
+class AllRefsDataclass(DataclassABC):
+    maybe_ref_value: str | Reference[str]
+
+
+@dataclasses.dataclass(frozen=True, kw_only=True)
+class MixedRefsDataclass(AllRefsDataclass, NoRefsDataclass):
+    ...
+
+
+@reference("refs::dc::NoRefsDataclassReference")
+class NoRefsDataclassReference(Reference[NoRefsDataclass]):
+    def __init__(self) -> None:
+        """
+        :param non_ref_value: The value
+        """
+        super().__init__()
+
+    def resolve(self, logger: LoggerABC) -> NoRefsDataclass:
+        return NoRefsDataclassReference()
+
+    def __str__(self):
+        return "NoRefsDataclassReference"
+
+
+class AllRefsDataclassReferenceABC[D: AllRefsDataclass](Reference[D]):
```

Comment:
Can the type extraction logic handle this? Doesn't this turn into any?

(after reading on: clearly not)

Reply:
I'm not sure which part exactly you were referring to. But I have the impression that you answered your own question, and that the test coverage suffices to give confidence?